core/providers/java/maven.go

```diff
@@ -0,0 +1,37 @@
+package java
+
+import (
+	"strings"
+
+	"github.com/railwayapp/railpack/core/generate"
+)
+
+const MAVEN_CACHE_KEY = "maven"
+
+func (p *JavaProvider) getMavenExe(ctx *generate.GenerateContext) string {
+	if ctx.App.HasMatch("mvnw") && ctx.App.HasMatch(".mvn/wrapper/maven-wrapper.properties") {
+		return "./mvnw"
+	}
+
+	return "mvn"
+}
+
+func (p *JavaProvider) mavenCache(ctx *generate.GenerateContext) string {
+	return ctx.Caches.AddCache(MAVEN_CACHE_KEY, ".m2/repository")
+}
+
+func getMavenPortConfig(ctx *generate.GenerateContext) string {
+	pomFile, err := ctx.App.ReadFile("pom.xml")
+
+	if err != nil {
+		return ""
+	}
+
+	if strings.Contains(pomFile, "<groupId>org.wildfly.swarm") {
+		return "-Dswarm.http.port=$PORT"
+	} else if strings.Contains(pomFile, "<groupId>org.springframework.boot") &&
+		strings.Contains(pomFile, "<artifactId>spring-boot") {
+		return "-Dserver.port=$PORT"
+	}
```

Comment:
as someone who hasn't used the java ecosystem in a really long time, this is so confusing to me 🙃
Can you maybe add some comments (here and in the above places where lots of random files are read) that explains the decisions being made. Otherwise in a year it will be forgotton